### PR TITLE
[baseline][libssh] Update to latest commit

### PR DESCRIPTION
**Pull request by LilyWangLL**
**Describe the pull request**

- #### What does your PR fix?
  Update `libssh` to latest commit, fix `libssh` fetch error on Linux.
```
/usr/bin/git fetch https://git.libssh.org/projects/libssh.git e8322817a9e5aaef0698d779ddd467a209a85d85 --depth 1 -n
error: Server does not allow request for unadvertised object e8322817a9e5aaef0698d779ddd467a209a85d85
```
All features test pass on x64-windows.

### CODE REVIEW

**Review by Cheney-W — DISMISSED** [review]

File: ports/libssh/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "libssh",
-  "version": "0.10.4",
+  "version-date": "2022-11-22",
```
Comment (dg0yt) [inline ports/libssh/vcpkg.json:3]: IIUC this change breaks version comparison and may not reach users in manifest mode. 
What about using `0.10.4+20221122`?

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/libssh/vcpkg.json (hunk 1 above)
Comment (LilyWangLL) [reply ports/libssh/vcpkg.json:3]: Is this `"version-string": "0.10.4+20221122"`?

Comment (LilyWangLL) [reply ports/libssh/vcpkg.json]: Done.

Comment (dg0yt) [reply ports/libssh/vcpkg.json]: > Is it `"version-string": "0.10.4+20221122"`?

No, it is still `version`. Already used in port `qwt`.

**Comment by LilyWangLL** [comment]
The upstream has latest commit yesterday, so this port install failed on Linux again. I changed to fetch latest commit. Do we have method to handle this problem?
cc @BillyONeal

**Comment by Neumann-A** [comment]
> The upstream has latest commit yesterday, so this port install failed on Linux again. I changed to fetch latest commit. Do we have method to handle this problem?

Yes we do. It is called `FETCH_REF master` (or however the libssh branch is called)

**Comment by LilyWangLL** [comment]
> > The upstream has latest commit yesterday, so this port install failed on Linux again. I changed to fetch latest commit. Do we have method to handle this problem?
> 
> Yes we do. It is called `FETCH_REF master` (or however the libssh branch is called)

Done, thanks for your help.

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**